server/api/api/utils.py

```diff
@@ -773,6 +774,61 @@ def ensure_function_has_auth_set(
             function.set_env_from_secret(env_key, secret_name, secret_key)
 
 
+def create_secret_env_vars_for_pipeline(
+    auth_info: mlrun.common.schemas.AuthInfo,
+) -> list[V1EnvVar]:
+    """
+    :param auth_info:   The auth info of the request.
+    """
+
+    if not auth_info.access_key:
+        auth_info.access_key = (
+            server.api.utils.auth.verifier.AuthVerifier().get_or_create_access_key(
+                session=auth_info.get_key(),
+                planes=[
+                    server.api.utils.clients.iguazio.SessionPlanes.data,
```

Comment:
Why `data` plane? Is it enough for the workflow's purposes? Doesn't it also need it for control-plane?

Reply:
From what I can tell this seems to be enough, I'm not sure why we would need the control plane as well for the workflow.